proxy/context.go

```diff
@@ -133,13 +136,14 @@ func newContext(
 	p *Proxy,
 ) *context {
 	c := &context{
-		responseWriter: w,
-		request:        r,
-		stateBag:       make(map[string]interface{}),
-		outgoingHost:   r.Host,
-		metrics:        &filterMetrics{impl: p.metrics},
-		proxy:          p,
-		routeLookup:    p.routing.Get(),
+		responseWriter:     w,
+		responseController: http.NewResponseController(w),
```

Comment:
It willnot be used for most of the requests so we can make ResponseController() return http.NewResponseController(w) on the fly

Reply:
Do you mean `func (*context) ResponseController() *http.ResponseController`?

Reply:
done